### ISSUE TITLE
Decorate throws `We don't know what to do with this node type.`

### ISSUE THREAD

**Issue by amilajack** [codemix/flow-runtime#28]
When setting decorate to true, the following error is throw:

```
someFileHere.js: We don't know what to do with this node type. We were previously a Statement but we can't fit in here?
```

`.babelrc`
```
{
  "presets": ["es2015", "stage-0"],
  "plugins": [
    "syntax-flow",
    ["flow-runtime", {
      "assert": true,
      "decorate": true
    }],
    "transform-flow-strip-types",
    "transform-runtime",
    "add-module-exports",
    "transform-es2015-modules-umd"
  ]
}
```

Setting decorators to false fixes the issue

**Comment by phpnode** [comment]
@amilajack please could you try and narrow down a reduced test case from the file that causes the error?

**Comment by amilajack** [comment]
Sure thing. I'll look into this more over the weekend

**Comment by dprokop** [comment]
I run into the same issue in my codebase that I try to introduce flow-runtime to. I have created a minimal babel/flow/flow-runtime setup to reproduce the issue:

```js 
// test.js
// @flow
export default function test(id : number) : number {
  return id
}
```

```js
// .babelrc
{
  "presets": [
    'react'
  ],
  "plugins": [
    ["flow-runtime", {
      assert: true,
      decorate: true
    }]
  ]
}
```

The stack trace for the error is: 
```js
Error: test.js: We don't know what to do with this node type. We were previously a Statement but we can't fit in here?
    at NodePath.insertAfter (/Users/prokopd/dev/experiments/node_modules/babel-traverse/lib/path/modification.js:175:13)
    at exit (/Users/prokopd/dev/experiments/node_modules/babel-plugin-flow-runtime/lib/transformVisitors.js:445:16)
    at NodePath._call (/Users/prokopd/dev/experiments/node_modules/babel-traverse/lib/path/context.js:76:18)
    at NodePath.call (/Users/prokopd/dev/experiments/node_modules/babel-traverse/lib/path/context.js:48:17)
    at NodePath.visit (/Users/prokopd/dev/experiments/node_modules/babel-traverse/lib/path/context.js:117:8)
    at TraversalContext.visitQueue (/Users/prokopd/dev/experiments/node_modules/babel-traverse/lib/context.js:150:16)
    at TraversalContext.visitSingle (/Users/prokopd/dev/experiments/node_modules/babel-traverse/lib/context.js:108:19)
    at TraversalContext.visit (/Users/prokopd/dev/experiments/node_modules/babel-traverse/lib/con
```

**Environment :**
`Node 6.5.0`
`Babel 6.18.0`
`flow-bin 0.37.4`

As @amilajack mentioned, the issue is present only when `decorate: true` is set.

**Comment by phpnode** [comment]
@dprokop thanks for this, I can replicate it and will push a fix today/tomorrow

**Comment by phpnode** [comment]
fixed in `babel-plugin-flow-runtime` v0.1.1
